### PR TITLE
Unit Test

### PR DESCRIPTION
**Pull request by odhiambo-ed**
## Description

In this milestone, we added unit tests to the codebase, specifically to the classes.

## Project requirements

- [ ] Add unit tests to your classes:
* Book
* Person
* Student
* Teacher
* Classroom
* Rental
* all classes that you created while implementing the Decorator design pattern

### CODE REVIEW

File: spec/person_spec.rb [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+  end
+
+  describe 'Can use services' do
+    context 'use method to validate age' do
+      it 'is of age' do
+        person = Person.new(18)
+        expect(person.of_age?).to be true
+      end
+    end
+  end
```
Comment (Mihndim2020) [inline spec/person_spec.rb:40]: - [ ] Hi team, great job so far with test coverage, but since we are aiming for a 100% test coverage, it will be nice for us to cover all the methods in all our classes with tests, taking into considerations both positive and negative test cases. For example, in this `Person class`, we could add the following test;
- Test for validating name: 
```
  describe "When validating name" do

    it "The validate_name method corrects the name" do
       `# * set up your test here
    end
  end
```
- Test for adding rentals: 
```ruby
  describe "When adding a new rentals" do
    
    it "The add_rental method returns a rental and adds its to the book" do
            `# * set up your test here
    end
  end
```
#### Please let's cover all our methods in all our classes with tests.

**Review by Mihndim2020 — CHANGES_REQUESTED** [review]
### Hi Team, 

### Good job so far!
### To Highlight! :clap:  :green_circle: 

:heavy_check_mark:  Good commit messages,
:heavy_check_mark: Descriptive Readme file,
:heavy_check_mark: `Linter checks` are green,

There are some issues that you still need to work on to get this project approved, but you are almost there!
ALMOST-MINION

# Status: Requires Changes ♻️
_Check the comments below under the review._

#### I will leave these comments here, since the files are not present for this milestone, as no changes were made on them 

- [ ] As it is a [requirement](https://github.com/microverseinc/curriculum-ruby/blob/main/oop/school_library_unit_tests.md) for this project, " `Add unit tests to your classes: all classes that you created while implementing the Decorator design pattern` ", please ensure all the methods in all classes that you created while implementing the Decorator design pattern are covered with tests as well.

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by Lordkaito — APPROVED** [review]
Hi @white3d, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 


## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._
